acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/tls/ClientSideTlsAcceptanceTest.java

```diff
@@ -0,0 +1,196 @@
+        TlsCertificateDefinition.loadFromResource("tls/cert2.pfx", "password2");
+
+    // Note: the HttpServer always responds with a JsonRpcSuccess, result=300.
+    final HttpServer web3ProviderHttpServer =
+        serverFactory.create(serverCert, ethSignerCert, workDir);
+
+    final Signer signer =
+        createAndStartSigner(
+            ethSignerCert, serverCert, web3ProviderHttpServer.actualPort(), 0, workDir);
+
+    assertThat(signer.accounts().balance("0x123456"))
+        .isEqualTo(BigInteger.valueOf(MockBalanceReporter.REPORTED_BALANCE));
+  }
+
+  @Test
+  void ethSignerDoesNotConnectToServerNotSpecifiedInTrustStore(@TempDir Path workDir)
+      throws IOException {
+    final TlsCertificateDefinition serverPresentedCert =
+        TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "password");
+    final TlsCertificateDefinition ethSignerCert =
+        TlsCertificateDefinition.loadFromResource("tls/cert2.pfx", "password2");
+    final TlsCertificateDefinition ethSignerExpectedServerCert =
+        TlsCertificateDefinition.loadFromResource("tls/cert2.pfx", "password2");
+
+    final HttpServer web3ProviderHttpServer =
+        serverFactory.create(serverPresentedCert, ethSignerCert, workDir);
+
+    final Signer signer =
+        createAndStartSigner(
+            ethSignerCert,
+            ethSignerExpectedServerCert,
+            web3ProviderHttpServer.actualPort(),
+            0,
+            workDir);
+
+    assertThatThrownBy(() -> signer.accounts().balance("0x123456"))
+        .isInstanceOf(ClientConnectionException.class)
+        .hasMessageContaining(String.format("%d", BAD_GATEWAY.code()));
+
+    // ensure submitting a transaction results in the same behaviour
+    final Transaction transaction =
+        Transaction.createEtherTransaction(
+            signer.accounts().richBenefactor().address(),
+            null,
+            GAS_PRICE,
+            INTRINSIC_GAS,
+            "0x1b00ba00ca00bb00aa00bc00be00ac00ca00da00",
+            Convert.toWei("1.75", Unit.ETHER).toBigIntegerExact());
+
+    assertThatThrownBy(() -> signer.transactions().submit(transaction))
+        .isInstanceOf(ClientConnectionException.class)
+        .hasMessageContaining(String.format("%d", BAD_GATEWAY.code()));
+  }
+
+  @Test
+  void missingKeyStoreForEthSignerResultsInEthSignerTerminating(@TempDir Path workDir)
+      throws IOException {
+    final TlsCertificateDefinition serverPresentedCert =
+        TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "password");
+    final TlsCertificateDefinition ethSignerCert =
+        new TlsCertificateDefinition(
+            workDir.resolve("Missing_keyStore").toFile(), "arbitraryPassword");
+
+    // Ports are arbitrary as EthSigner should exit
+    final Signer signer = createSigner(ethSignerCert, serverPresentedCert, 9000, 9001, workDir);
+    signer.start();
+    waitFor(() -> assertThat(signer.isRunning()).isFalse());
```

Comment:
Is there any chance of a race condition here?

Reply:
don't _think_ so - start is guaranteed to not return before the process is running, we just want to make sure its also stopped.